### PR TITLE
commit: support commit.verbose and --no-verbose

### PR DESCRIPTION
**Pull request by calebhearth**
This patch allows people to set commit.verbose to implicitly send
--verbose to git-commit. It also introduces --no-verbose to override the
configuration setting.

It introduces several cleanup patches to t/t7505-commit-verbose.sh to
bring it closer to the current state of the tests as they have been
explained them to me, then adds the verbose config and --no-verbose
flag.

Since the last version of this patch
(http://marc.info/?l=git&m=140113063517118&w=2), I've made the following
changes:
- Remove the $PWD => $(pwd) change, as it was a misunderstanding of
  t/README (Thank you to Johannes and David for pointing that out and
  clarifying)
- Use consistent test naming (Thanks to Jeremiah)
- Switch from <<-EOF to <<-'EOF' without manual escaping in write_script
  calls for the test editors check-for-diff and check-for-no-diff. The
  proposed <<-\EOF syntax did not properly prevent the parameter
  substitution on my machine, as both Eric and the documentation
  suggested it should, but the quoted version seems to work fine.
  (Thanks to Eric)
- Fix extra test_set_editor invocations and broken %%-chains created as
  a result of the test_set_editor changes. (Thanks again to Eric)

### CODE REVIEW

**Comment by calebhearth** [comment]
http://marc.info/?l=git&m=140251155830422&w=2
http://marc.info/?l=git&m=140251156130423&w=2
http://marc.info/?l=git&m=140251156530427&w=2
http://marc.info/?l=git&m=140251156230424&w=2
http://marc.info/?l=git&m=140251156730429&w=2

File: t/t7507-commit-verbose.sh [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
 
-cat >check-for-diff <<EOF
-#!$SHELL_PATH
-exec grep '^diff --git' "\$1"
+write_script check-for-diff <<-'EOF'
+	exec grep '^diff --git' "$1"
+EOF
+
+write_script check-for-no-diff <<-EOF
+	exec grep -v '^diff --git' "\$1"
```
Comment (mike-burns) [inline t/t7507-commit-verbose.sh]: There is an inconsistency between `check-for-diff`, which uses `'EOF'`, and `check-for-no-diff`, which uses a normal `EOF` but then escapes the `$`.

Comment (calebhearth) [inline t/t7507-commit-verbose.sh]: Good catch, thank you.

**Comment by calebhearth** [comment]
[[PATCH v5 0/4] commit: Add commit.verbose configuration](http://marc.info/?t=140260211100001&r=1&w=2)
[ [PATCH v5 1/4] commit test: Use test_config instead of git-config](http://marc.info/?l=git&m=140260197027501&w=2)
[[PATCH v5 2/4] commit test: Use write_script](http://marc.info/?t=140260211100004&r=1&w=2)
[[PATCH v5 3/4] commit test: test_set_editor in each test](http://marc.info/?t=140260211100002&r=1&w=2)
[[PATCH v5 4/4] commit: Add commit.verbose configuration](http://marc.info/?l=git&m=140260324228027&w=2)

**Comment by calebhearth** [comment]
http://thread.gmane.org/gmane.comp.version-control.git/251943

**Pull request closed without merging**